Pilot/pilotCommands.py

```diff
@@ -92,16 +90,28 @@ def wrapper(self):
             self.log.info(
                 "Flushing the remote logger buffer for pilot on sys.exit(): %s (exit code:%s)" % (pRef, str(exCode))
             )
-            self.log.buffer.flush()  # flush the buffer unconditionally (on sys.exit()).
-            try:
-                sendMessage(self.log.url, self.log.pilotUUID, self.log.wnVO, "finaliseLogs", {"retCode": str(exCode)})
-            except Exception as exc:
-                self.log.error("Remote logger couldn't be finalised %s " % str(exc))
+            if self.pp.jwt:
+                try:
+                    self.log.error(str(exCode))
+                    self.log.error(traceback.format_exc())
+                    self.log.buffer.flush(force=True)
+                except Exception as exc:
+                    self.log.error("Remote logger couldn't be finalised %s " % str(exc))
+            else:
+                # No force here because there's no remote logger if we're here
```

Comment:
Not sure what you mean here. You are clearly within remote logger (if the remote logger is not enabled the wrapper returns at line 80). And the classic logger has no buffer...